src/docfx/build/page/PageBuilder.cs

```diff
@@ -261,6 +266,27 @@ private SystemMetadata CreateSystemMetadata(ErrorBuilder errors, FilePath file,
         return systemMetadata;
     }
 
+    private string? GetCanonicalUrlWithMonikerIfNecessary(string? pageType, FilePath file)
```

Comment:
There is duplicate code between `GetCanonicalUrlWithMonikerIfNecessary` and `GetCanonicalUrl`. How about defining this API as `AppendCanonicalMonikerIfNecessary`?